### PR TITLE
feat(FEC-13875): Child entries download

### PR DESCRIPTION
**Pull request by semarche-kaltura**
### Description of the Changes

resolves: https://kaltura.atlassian.net/browse/FEC-13875

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: translations/en.i18n.json [hunk 1]
```diff
@@ -14,7 +14,10 @@
       "source_label": "Source",
       "more_captions_label": "More captions",
       "less_captions_label": "Less captions",
-      "attachments_label": "Attachments"
+      "attachments_label": "Attachments",
+      "main_stream": "Main stream",
+      "additional_streams": "Additional streams",
+      "captions": "Captions"
```
Comment (semarche-kaltura) [inline translations/en.i18n.json:20]: translates ticket
https://kaltura.atlassian.net/browse/FEC-14247

File: src/components/download-overlay/download-overlay.tsx [hunk 2]
```diff
@@ -28,98 +28,139 @@ interface DownloadOverlayProps extends AssetsListProps {
...
+        additionalStreams,
+        captionsLabel,
+        downloadMetadatas,
+        updateOverlay
+      }: DownloadOverlayProps) => {
+        const [isVisible, setIsVisible] = useState(false);
+        const mainSourceMetadata = downloadMetadatas[0];
+        // tslint:disable-next-line: ts-expect-error
+        // @ts-expect-error: Property 'config' is protected and only accessible within class 'BasePlugin' and its subclasses
+        const downloadConfig = downloadPluginManager.downloadPlugin.config;
```
Comment (SivanA-Kaltura) [inline src/components/download-overlay/download-overlay.tsx:66]: at this point, maybe just pass config inside a context if we want it to be accessible to components at more inner levels

File: src/download-plugin-manager.tsx [hunk 3]
```diff
@@ -13,28 +13,40 @@ const {FakeEvent} = core;
...
+  constructor(public downloadPlugin: Download, private _config: DownloadConfig, private _logger: any, private _eventManager: core.EventManager) {
     super();
-    this.downloadService = new DownloadService(this.downloadPlugin.player, this._logger);
+    this.downloadService = new DownloadService(this.downloadPlugin.player, this._logger, this._eventManager);
   }
 
-  async getDownloadMetadata(refresh = false): Promise<DownloadMetadata> {
-    if (!this.downloadMetadata || refresh) {
-      this.downloadMetadata = await this.downloadService.getDownloadMetadata(this._config);
+  isMetadataEmpty(metadata: DownloadMetadata): boolean {
```
Comment (SivanA-Kaltura) [inline src/download-plugin-manager.tsx:25]: why does this need a separate function ? do you think there will be additional logic here in the future ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/download-plugin-manager.tsx (hunk 3 above)
Comment (semarche-kaltura) [reply src/download-plugin-manager.tsx:25]: `isMetadataEmpty` uses it in 2 places, that's why check moved to separate method:
https://github.com/kaltura/playkit-js-downloads/pull/78/files#diff-3194df8725001bb4b9724e8da79b6078668accf54d61dfe6bdb01b9d8aa3953aR30
https://github.com/kaltura/playkit-js-downloads/pull/78/files#diff-3194df8725001bb4b9724e8da79b6078668accf54d61dfe6bdb01b9d8aa3953aR41

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**